zerver/models/users.py

```diff
@@ -884,6 +886,10 @@ def get_user_profile_by_id(user_profile_id: int) -> UserProfile:
         "realm__can_access_all_users_group__named_user_group",
         "realm__can_create_public_channel_group",
         "realm__can_create_public_channel_group__named_user_group",
+        "realm__direct_message_initiator_group",
+        "realm__direct_message_initiator_group__named_user_group",
```

Comment:
`get_user` function in this file is used to get the sender object for sending mirrored messages, so might be good to prefetch these fields there as well. Though I am not sure how performance sensitive is the code for sending mirrored messages and if it would be helpful to avoid these extra queries.